### ISSUE TITLE
Add Passthrough Mode

### ISSUE THREAD

**Issue by Schmetzler** [PhoneVR-Developers/PhoneVR#269]
Are there any plans on supporting pass through?
As the camera is available this should be possible... I implemented a passthrough within google cardboard sdk 
and would like to contribute to the repo. But I do not know if something is already present?

Besides I would need a little research to trigger passthrough mode via accelerator sensors (like it is triggered in the quest).

If there is some interest in this I would dig into it.

**Comment by ShootingKing-AM** [comment]
I hope you are saying that you would like to display Camera Image in ALVR lobby or something similar.
If thats so, there are no plans for this. 

PRs are welcome though. I had this thoughts in my mind but it didnt even make to planning stage. This would be a good feature though :)

**Comment by Lamonz88** [comment]
Yes. It would be amazing to have this feature because our phones have much better cameras

**Comment by Lamonz88** [comment]
> I hope you are saying that you would like to display Camera Input in ALVR lobby or something similar. If thats so, there are no plans for this.
> 
> PRs are welcome though. I had this thoughts in my mind but it didnt even make to planning stage. This would be a good feature though :)

Is it actually possible to implement this feature in future updates. 

I'd rather have my phone than buy another VR headst because my phone's camera is much better ..

**Comment by ShootingKing-AM** [comment]
Yup it's possible, wip. Need more time or need more coffee so that I can lure in more developers to work on this :3

**Comment by Schmetzler** [comment]
I already experimented with it... And had success.... But I am currently trying to build this repo, so that I can insert this code and test it... But as I am not really fond of Android development and stuck in dependency hell... It may take a while. If I have some spare time and can get around those dependency issues I would gladly contribute.

**Comment by ShootingKing-AM** [comment]
What issues are you facing, you can text on #dev channel in discord for some instant help.

**Comment by Lamonz88** [comment]
> I already experimented with it... And had success.... But I am currently trying to build this repo, so that I can insert this code and test it... But as I am not really fond of Android development and stuck in dependency hell... It may take a while. If I have some spare time and can get around those dependency issues I would gladly contribute.

Are looking fur funding or sponsors..? 

If so , maybe you can set up donations or give is option to sponsor you on your repo.!

**Comment by Lamonz88** [comment]
> Yup it's possible, wip. Need more time or need more coffee so that I can lure in more developers to work on this :3

Thank you soo much fir your reply..

That would we great if it's possible..

**Comment by ShootingKing-AM** [comment]
> > I already experimented with it... And had success.... But I am currently trying to build this repo, so that I can insert this code and test it... But as I am not really fond of Android development and stuck in dependency hell... It may take a while. If I have some spare time and can get around those dependency issues I would gladly contribute.
> 
> Are looking fur funding or sponsors..?
> 
> If so , maybe you can set up donations or give is option to sponsor you on your repo.!

Yes am looking for funding and sponsors, since the work is rising and no new developers are joining.

It's already done? Or do you mean GitHub sponsors ? Is there any benefit of GitHub sponsors over open collective?

**Comment by Lamonz88** [comment]
> It's already done? Or do you mean GitHub sponsors ? Is there any benefit of GitHub sponsors over open collective?

Yea. Do you have your own branch or repo with Pronsorship enabled ..?.

I guess a Patron would be really good to have .

But I wouldn't mind sending you a donation to your PayPal as a friend..

**Comment by Schmetzler** [comment]
So there is a passthrough mode available on my fork: Schmetzler/PhoneVR@a215d89.
Currently it is just available via the dropdown menu on the upper right... which is not really usable.
Next I will try to implement a gesture such as tapping the side of the headset to toggle the passthrough mode.

**Comment by ShootingKing-AM** [comment]
You can make a PR to this repo. Even basic pass through is very helpful

**Comment by Schmetzler** [comment]
So added the PR #313, now also with the experimental double tap detection... Well this was harder than anticipated, but it works somehow. If anyone has another suggestion how the pass-through could be detected, please let me (or us ) know.
